Robust.Shared/Serialization/TypeSerializers/Implementations/TimespanSerializer.cs

```diff
@@ -9,42 +10,98 @@
-            IDependencyCollection dependencies, bool alwaysWrite = false,
-            ISerializationContext? context = null)
+    public ValidationNode Validate(
+        ISerializationManager serializationManager,
+        ValueDataNode node,
+        IDependencyCollection dependencies,
+        ISerializationContext? context = null)
+    {
+        return TryTimeSpan(node, out _)
+            || double.TryParse(node.Value, NumberStyles.Any, CultureInfo.InvariantCulture, out _)
+            ? new ValidatedValueNode(node)
+            : new ErrorNode(node, "Failed parsing TimeSpan");
+    }
+
+    public DataNode Write(
+        ISerializationManager serializationManager,
+        TimeSpan value,
+        IDependencyCollection dependencies,
+        bool alwaysWrite = false,
+        ISerializationContext? context = null)
+    {
+        return new ValueDataNode(value.TotalSeconds.ToString(CultureInfo.InvariantCulture));
+    }
+
+    [MustUseReturnValue]
+    public TimeSpan CreateCopy(
+        ISerializationManager serializationManager,
+        TimeSpan source,
+        IDependencyCollection dependencies,
+        SerializationHookContext hookCtx,
+        ISerializationContext? context = null)
+    {
+        return source;
+    }
+
+    /// <summary>
+    /// Convert strings from the compatible format (or just numbers) into TimeSpan and output it. Returns true if successful.
+    /// The string must start with a number and end with a single letter referring to the time unit used.
+    /// It can NOT combine multiple types (like "1h30m"), but it CAN use decimals ("1.5h")
+    /// </summary>
+    private bool TryTimeSpan(ValueDataNode node, [NotNullWhen(true)] out TimeSpan? timeSpan)
+    {
+        timeSpan = null;
+
+        // A lot of the checks will be for plain numbers, so might as well rule them out right away, instead of
+        // running all the other checks on them. They will need to get parsed later anyway, if they weren't now.
+        if (double.TryParse(node.Value, out var v))
         {
-            return new ValueDataNode(value.TotalSeconds.ToString(CultureInfo.InvariantCulture));
+            timeSpan = TimeSpan.FromSeconds(v);
+            return true;
         }
 
-        [MustUseReturnValue]
-        public TimeSpan CreateCopy(ISerializationManager serializationManager, TimeSpan source,
-            IDependencyCollection dependencies, SerializationHookContext hookCtx, ISerializationContext? context = null)
+        // If there aren't even enough characters for a number and a time unit, exit
+        if (node.Value.Length <= 1)
+            return false;
+
+        // If the input without the last character is still not a valid number, exit
+        if (!double.TryParse(node.Value.AsSpan()[..^1], out var number))
+            return false;
+
+        // Check the last character of the input for time unit indicators
+        switch (node.Value.AsSpan()[^1])
```

Comment:
You don't need to `.AsSpan()` this since you're just indexing to get a single char out.
```suggestion
        switch (node.Value[^1])
```